src/framework/smoke_job.hpp

```diff
@@ -154,9 +154,10 @@ class SmokeJob {
     return task;
   }
 
-  static std::list<SmokeJob> createJobsFromJson(const SmokeFlags flags, bool&
+  static std::list<std::shared_ptr<SmokeJob>> createJobsFromJson(
```

Comment:
Do you have reason to use shared_ptrs? If not, please change it to unique.
Tracking down memory leaks from shared pointers scattered in whole codebase is not fun.